File: phaseGateProject/MenstrualApp.java
import java.time.LocalDate;
import java.util.Scanner;
import java.time.format.DateTimeFormatter;



public class MenstrualApp{


	public static void main (String[] args){

	

	getMenu();

	}

	public static void getMenu(){
	Scanner scan = new Scanner(System.in);
	String menu = """
		          ***********************************
	
		          >> 1 Check Next Menstrual Period
		          >> 2 Check Ovulation Date	
		          >> 3 Check fertity period
		           ***********************************	
						                 """;
		System.out.print(menu);

	
try{
	System.out.print("Enter your number of choice please:  ");
	int userInput = scan.nextInt();



	if(userInput == 1){
		System.out.print("Enter last period full details in this format (\"dd/MM/yyyy\"): ");

		String lastPeriodDay = scan.next();
		

		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy"); 
 
		LocalDate date = LocalDate.parse(lastPeriodDay, formatter);

		System.out.print("Enter length cycle:  ");

		int lengthCycle = scan.nextInt();
		
		 if (lengthCycle < 21 || lengthCycle > 35){
			System.out.print("Please see a doctor this is not a normal cycle length");
		
		 }else{
			LocalDate nextPeriodDate = date.plusDays(lengthCycle);
		          System.out.print("Your next menstrual date is: " +nextPeriodDate);
	
		}
		}

	if(userInput == 2){


		System.out.print("Enter last period full details in this format (\"dd/MM/yyyy\"): ");

		String lastPeriodDay = scan.next();

		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy"); 
 
		LocalDate date = LocalDate.parse(lastPeriodDay, formatter);

		System.out.print("Enter length cycle ");

		int lengthCycle = scan.nextInt();

		int Ovulation = lengthCycle / 2;
		
		LocalDate dateOfOvulation = date.plusDays(Ovulation);

		System.out.print("your Ovulation date is: " +dateOfOvulation);
		
		}
		
		
	if(userInput == 3){
		System.out.print("Enter length cycle ");

		int lengthCycle = scan.nextInt();

		int Ovulation = lengthCycle / 2;

		int fertitilityStart = Ovulation - 3 ;
		
		int fertitilityEnd = Ovulation + 3;

		
		System.out.println("your fertility period Start:" +fertitilityStart);
		System.out.print("your fertility period Ends:" +fertitilityEnd);
		
		

	}

}
catch(Exception e){
	System.out.println("Please check your input and put in the correct thing");
}
		


}



}

File: gateOne/MenstrualApp.py
import datetime

def get_menu():
    menu = """
    ***********************************

    >> 1 Check Next Menstrual Period
    >> 2 Check Ovulation Date    
    >> 3 Check Fertility Period
    ***********************************
    """
    print(menu)

    try:
        user_input = int(input("Enter your number of choice please: "))

        if user_input == 1:
            last_period_day = input("Enter last period full details in this format (\"dd/MM/yyyy\"): ")
            date = datetime.datetime.strptime(last_period_day, "%d/%m/%Y").date()
            length_cycle = int(input("Enter length cycle: "))

            if length_cycle < 21 or length_cycle > 35:
                print("Please see a doctor; this is not a normal cycle length")
            else:
                next_period_date = date + datetime.timedelta(days=length_cycle)
                print("Your next menstrual date is:", next_period_date)

        elif user_input == 2:
            last_period_day = input("Enter last period full details in this format (\"dd/MM/yyyy\"): ")
            date = datetime.datetime.strptime(last_period_day, "%d/%m/%Y").date()
            length_cycle = int(input("Enter length cycle: "))
            ovulation = length_cycle // 2
            date_of_ovulation = date + datetime.timedelta(days=ovulation)
            print("Your ovulation date is:", date_of_ovulation)

        elif user_input == 3:
            length_cycle = int(input("Enter length cycle: "))
            ovulation = length_cycle // 2
            fertility_start = ovulation - 3
            fertility_end = ovulation + 3
            print("Your fertility period starts:", fertility_start)
            print("Your fertility period ends:", fertility_end)

    except Exception:
        print("Please check your input and enter the correct format")

get_menu()
